import Function_Anonymous_Run (add2ToAllListElement)
import qualified TechIO.Helper as TechIo
import qualified TechIO.VerifyUsedFunctionSimple as Verify
import Control.Monad
import Control.Exception
import Data.IORef

currentTest = do
    TechIo.simpleTest (add2ToAllListElement [1..5]) [3..7]
    TechIo.simpleTest (add2ToAllListElement [1..20]) [3..22]

main = do
  result <- TechIo.simpleRunner currentTest

  case result of
    TechIo.Failure -> do
      TechIo.sendMsg "Astuce üí°" "Humain √† ce rythme je vais te battre \x1F914"
      TechIo.failure
    TechIo.Success -> do
      used <- Verify.isFunctionUsed
      case used of
        True -> do
          TechIo.sendMsg "Bien jou√©. üéÜ" "Toujours plus vite toujours plus loin"
          TechIo.success
        False -> do
          TechIo.sendMsg "Oops! üêû" "N'essayes pas de m'avoir humain, utilise la fonction map!"
          TechIo.failure
